python/taichi/lang/transformer.py

```diff
@@ -736,7 +748,15 @@ def visit_Assert(self, node):
     def visit_Return(self, node):
         self.generic_visit(node)
         if self.is_kernel:
-            raise TaichiSyntaxError(
-                '"return" not allowed in \'ti.kernel\'. Please walk around by storing the return result to a global variable.'
-            )
+            # todo: we only support return at the end of kernel, check this
```

Comment:
```suggestion
            # TODO: we only support return at the end of a kernel. The following code does a check for this.
```

Reply:
Does the following code check if the return statement is at the end of a kernel?

Reply:
It doesn't for now, it's a TODO item.